packages/core/src/components/graphs/gauge.ts

```diff
@@ -287,10 +294,14 @@ export class Gauge extends Component {
 				.merge(deltaNumber)
 				.attr('text-anchor', 'middle')
 				.style('font-size', `${deltaFontSize(radius)}px`)
-				.text(
-					(d: any) =>
-						`${localeNumberFormatter(Number(numberFormatter(d)), localeCode)}${gaugeSymbol}`
-				)
+				.text((d: any) => {
+					const value = Number(d.toFixed(2)) % 1 !== 0 ? d.toFixed(2) : d.toFixed()
+					if (numberFormatter) {
+						return `${numberFormatter(value)}${gaugeSymbol}`
+					} else {
+						return `${localeNumberFormatter(Number(value), localeCode)}${gaugeSymbol}`
```

Comment:
is there ever a chance of `localeNumberFormatter` being null or undefined? we should have a fallback probably

Reply:
Do you mean the value passed to the localeNumberFormatter or the return value of localeNumberFormatter? can you please clarify.